frontend/system/src/Entities/ListView.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from '@common/intl';
+import {
+  AppBarContentPortal,
+  DataTable,
+  NothingHere,
+  TableProvider,
+  createTableStore,
+  useColumns,
+  SearchToolbar,
+  ToggleButton,
+} from '@common/ui';
+import { useBreadcrumbs, useQueryParamsState } from '@common/hooks';
+import { useEntities } from '../api';
+import { ToggleButtonGroup } from '@mui/material';
+
+export const ListView = () => {
+  const t = useTranslation('system');
+  const { setSuffix } = useBreadcrumbs();
+
+  useEffect(() => {
+    setSuffix('Browse');
+  }, []);
+
+  const { filter, queryParams, updatePaginationQuery, updateSortQuery } =
+    useQueryParamsState({
+      initialSort: { key: 'description', dir: 'asc' },
+    });
+
+  const { sortBy, page, offset, first } = queryParams;
+
+  const columns = useColumns(
+    [
+      { key: 'code', label: 'label.code', width: 200, sortable: false },
+      { key: 'description', label: 'label.description', width: 1000 },
+      { key: 'type', label: 'label.type', sortable: false, width: 200 },
+    ],
+    { sortBy: sortBy, onChangeSortBy: updateSortQuery },
+    [sortBy, updateSortQuery]
+  );
+
+  const [categories, setCategories] = useState<string[]>(['drug']);
+
+  const searchFilter = filter.filterBy?.['search'];
+  const search = typeof searchFilter === 'string' ? searchFilter : '';
+
+  const { data, isError, isLoading } = useEntities({
+    filter: {
+      categories,
+      description: search,
+      orderBy: {
+        field: sortBy.key,
+        descending: sortBy.isDesc,
+      },
+    },
+    first,
+    offset,
+  });
```

Comment:
The `useQueryParams` hook we provide from our `common` package returns are very specific structure, targeted for the filters in our Rust backend (and the `useColumns`/`DataTable` components in the frontend). This backend is a little different, so instead of being able to simply do `useEntities(queryParams)`, I have to do the above mapping (finding the `searchFilter` piece particularly unappealing.)

I thought about implementing a new `useQueryParams` hook specific for this repo? But then wasn't sure how set we are on this different structure from the backend perspective, particular given half of the fields on `filter` aren't even being handled yet...

Reply:
Will add a ticket to circle back on this 👍